src/python/espressomd/particle_data.pyx

```diff
@@ -151,6 +151,8 @@ cdef class ParticleHandle:
 
         def __set__(self, _pos):
             cdef double mypos[3]
+            if np.isnan(pos).any() or np.isinf(pos).any():
```

Comment:
Why do you want to check that? I'm not sure that I follow. What scenario is this check for?

Reply:
it seems that the callbacks do not handle ```nan```. Please try:
``` python
import espressomd

system = espressomd.System(box_l=[10.0]*3)
system.part.add(pos=[float('nan')]*3)     

```

Reply:
since ```pos``` is the only particle property where the target node depends on the value of the property we should validate before calling the callback.